db/migrate/20250717003848_create_username_blocks.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class CreateUsernameBlocks < ActiveRecord::Migration[8.0]
+  def change
+    create_table :username_blocks do |t|
+      t.string :username, null: false
+      t.string :normalized_username, null: false
+      t.boolean :exact, null: false, default: false
+      t.boolean :allow_with_approval, null: false, default: false
+
+      t.timestamps
+    end
+
+    add_index :username_blocks, 'lower(username)', unique: true, name: 'index_username_blocks_on_username_lower_btree', order: { username: :asc }
```

Comment:
Why is this on `lower(username)` and then sorted on `username`?

Reply:
Would I have to write `ORDER BY lower(username) ASC` in the query for it to use this index?

I thought we might be able to use the index for sorting via `ORDER BY username ASC`.

Reply:
Actually, the index is only used if there is a `WHERE` clause, so never mind this, we don't need it.